src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -512,6 +515,36 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Allows the database type to be customized if we
+        /// are not able to properly infer the database type.
+        /// </summary>
+        /// <param name="dbDataParameter">The parameter.</param>
+        /// <param name="parameterInfo"></param>
+        internal void CustomizeParameterType(IDbDataParameter dbDataParameter, DatabaseParameterInfo parameterInfo)
```

Comment:
Hmm, I'm not quite understanding this one. We are working with the database column type not the value here. What do you think about "SetParamDbType" for the method name?